### DISCUSSION TITLE
How to decode json fields with mixed case

### DISCUSSION THREAD

**Discussion by amichelins** [vlang/v#11545] [General]
I have this json from a external API
`
[
    {
        "number": "674952",
        "createdAt": "2021-01-26 20:38:47",
        "updatedAt": "2021-02-18 14:51:33",
        "status": {
            "name": "status.returned"
        },
    }
]
`
This fields updatedAt, createdAtgive my erros at compile time.
this is my struct:
`
struct Pedido {
    number string
    status Status
    createAt string
    updateAt string
}
`
json.v:9:5: error: field name `createAt` cannot contain uppercase letters, use snake_case instead

How to decode this fields?

Annotations are allow in vlang?
Go allows this
`
struct Pedido {
    number string 
    status Status 
    CreateAt string `json:"createAt"`
    UpdateAt string  `json:"updateAt"`
}
`
This `json:"createAt"` tell go that createAt field in json is map to  CreateAt in the struct

**Comment by JalonSolov** [comment]
It's in the docs...
https://github.com/vlang/v/blob/master/doc/docs.md#decoding-json

Example:
```v
    last_name string [json: lastName]
```

**Reply by amichelins** [reply]
@JalonSolov Thank you very much.
I really did not look carefully

**Reply by amichelins** [reply]
Tested, working perfectly. And so fast!!!
